### PR TITLE
Updated armorblox plugin with the review comments.

### PR DESCRIPTION
**Pull request by SamhithaTatipalli**

### CODE REVIEW

File: plugins/armorblox/help.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+|Name|Type|Required|Description|
+|----|----|--------|-----------|
+|incidents|[]incident|True|A list of incidents identified by Armorblox|
+
+Example output:
+
+```
+{
+  "{\"incidents\": \"some incidents\"}"
+}
```
Comment (igorski-r7) [inline plugins/armorblox/help.md:103]: Please reformat it 😄

File: plugins/armorblox/icon_armorblox/actions/get_remediation_action/action.py
Comment (igorski-r7) [inline plugins/armorblox/icon_armorblox/actions/get_remediation_action/action.py:1]: Please add newline at the end of file

File: plugins/armorblox/icon_armorblox/connection/connection.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        self.api = None
+
+    def connect(self, params):
+        self.logger.info("Connect: Connecting...")
+        api_key = params.get(Input.API_KEY).get("secretKey")
+        tenant_name = params.get(Input.TENANT_NAME)
+        self.api = ArmorbloxAPI(api_key = api_key, tenant_name = tenant_name, logger=self.logger)
+
+    def test(self):
+
```
Comment (igorski-r7) [inline plugins/armorblox/icon_armorblox/connection/connection.py:21]: I think that extra space is not necessary 😄

File: plugins/armorblox/icon_armorblox/connection/connection.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+class Connection(insightconnect_plugin_runtime.Connection):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(input=ConnectionSchema())
+        self.api = None
+
+    def connect(self, params):
+        self.logger.info("Connect: Connecting...")
+        api_key = params.get(Input.API_KEY).get("secretKey")
```
Comment (igorski-r7) [inline plugins/armorblox/icon_armorblox/connection/connection.py:16]:
```suggestion
        api_key = params.get(Input.API_KEY, {}).get("secretKey")
```

File: plugins/armorblox/icon_armorblox/triggers/get_incidents/trigger.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    def __init__(self):
+        super(self.__class__, self).__init__(
+                name='get_incidents',
+                description=Component.DESCRIPTION,
+                input=GetIncidentsInput(),
+                output=GetIncidentsOutput())
+
+    def run(self, params={}):
+        """Run the trigger"""
+        fetch_interval = params.get(Input.INTERVAL)
```
Comment (igorski-r7) [inline plugins/armorblox/icon_armorblox/triggers/get_incidents/trigger.py:19]: Please specify the default interval value 
```suggestion
        fetch_interval = params.get(Input.INTERVAL, DEFAULT_INTERVAL_VALUE)
```

**Review by igorski-r7 — CHANGES_REQUESTED** [review]

File: plugins/armorblox/help.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
+# Description
+
+Armorblox secures enterprise communications over email and other cloud office applications with the power of Natural Language Understanding. The Armorblox platform connects over APIs and analyzes thousands of signals to understand the context of communications and protect people and data from compromise. Over 56,000 organizations use Armorblox to stop BEC and targeted phishing attacks, protect sensitive PII and PCI, and automate remediation of user-reported email threats. Armorblox was featured in the 2019 Forbes AI 50 list and was named a 2020 Gartner Cool Vendor in Cloud Office Security. Founded in 2017, Armorblox is headquartered in Sunnyvale, CA and backed by General Catalyst and Next47
```
Comment (cmcnally-r7) [inline plugins/armorblox/help.md:3]: Make sure the `help.md` description is identical to the `plugin.spec.yaml` description

File: plugins/armorblox/help.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+## Troubleshooting
+
+_This plugin does not contain any troubleshooting information._
+
+# Version History
+
+* 1.0.0 - Initial plugin
+
+# Links
+
```
Comment (cmcnally-r7) [inline plugins/armorblox/help.md:163]: Please move values in `references` to `links`.
Also please ensure any relevant links to armorblox api docs are included

File: plugins/armorblox/plugin.spec.yaml [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+      description: Is email tagged
+      type: boolean
+      required: false
+    date:
+      title: Incident Date
+      description: Date of the incident
+      type: date
+      required: false
+    policy_names:
+      title: policy_names
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:80]:
```suggestion
      title: Policy Names
```

File: plugins/armorblox/plugin.spec.yaml [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+      description: List of policies
+      type: '[]string'
+      required: false
+    title:
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:84]:
```suggestion
    subject:
```

File: plugins/armorblox/plugin.spec.yaml [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+      title: Subject
+      description: Mail subject
+      type: 'string'
+      required: false          
+    resolution_state:
+      title: resolution_state
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:90]:
```suggestion
      title: Resolution State
```

File: plugins/armorblox/plugin.spec.yaml [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+      description: Incident ID
+      type: string
+      required: false
+    app_name:
+      title: App Name
+      description: App Name
+      type: string
+      required: false 
+    external_senders:
+      title: External senders
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:110]:
```suggestion
      title: External Senders
```

File: plugins/armorblox/plugin.spec.yaml [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+      description: List of external senders
+      type: "[]string"
+      required: 
+    folder_categories:
+      title: Folder categories
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:115]:
```suggestion
      title: Folder Categories
```

File: plugins/armorblox/plugin.spec.yaml [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+      description: Engagements 
+      type: engagement
+      required: false
+    remediation_actions:
+      title: Remediation Action
+      description: Remediation Action
+      type: "[]string"
+      required: false
+    users:
+      title: users
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:140]:
```suggestion
      title: Users
```

File: plugins/armorblox/plugin.spec.yaml [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+      description: List of users
+      type: "[]user"
+      required: false
+    external_users:
+      title: External users
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:145]:
```suggestion
      title: External Users
```

File: plugins/armorblox/plugin.spec.yaml [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+      description: List of external users
+      type: "[]user"
+      required: false
+    final_detection_tags:
+      title: Detection tags
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:150]:
```suggestion
      title: Detection Tags
```

File: plugins/armorblox/unit_test/test_get_remediation_action.py
Comment (cmcnally-r7) [inline plugins/armorblox/unit_test/test_get_remediation_action.py:1]: Can we please use `parameterized` on these tests?
https://pypi.org/project/parameterized/

File: plugins/armorblox/plugin.spec.yaml [hunk 15]
```diff
@@ -0,0 +1,206 @@
...
+      type: string
+      required: false
+    is_vip:
+      title: Is User VIP
+      description: Is User VIP
+      type: boolean
+      required: false
+  final_detection_tag:
+    detection_tag_id:      
+      title: Detection tag ID
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:43]:
```suggestion
      title: Detection Tag ID
```

File: plugins/armorblox/plugin.spec.yaml [hunk 16]
```diff
@@ -0,0 +1,206 @@
...
+      description: Detection tag ID
+      type: string
+      required: false
+    detection_tag_name:
+      title: Detection tag name
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:48]:
```suggestion
      title: Detection Tag name
```

File: plugins/armorblox/plugin.spec.yaml [hunk 17]
```diff
@@ -0,0 +1,206 @@
...
+
+types:
+  user:
+    name:
+      title: User name
+      description: Name of the user
+      type: string
+      required: false
+    email:
+      title: User email
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:32]:
```suggestion
      title: User Email
```

File: plugins/armorblox/plugin.spec.yaml [hunk 18]
```diff
@@ -0,0 +1,206 @@
...
+resources:
+  source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/plugins/armorblox
+  license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url: https://armorblox.com/
+enable_cache: true
+
+types:
+  user:
+    name:
+      title: User name
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:27]:
```suggestion
      title: User Name
```

File: plugins/armorblox/plugin.spec.yaml [hunk 19]
```diff
@@ -0,0 +1,206 @@
...
+  tenant_name:
+    title: Tenant Name
+    description: Armorblox Tenant Name
+    type: string
+    required: true
+    example: my-tenant-name
+  api_key:
+    type: credential_secret_key
+    title: Armorblox API Key
+    description: Armorblox API Key
```
Comment (cmcnally-r7) [inline plugins/armorblox/plugin.spec.yaml:165]: This is purely optional, the functionality won't change here. It's just good practice to have the pattern of the fields stay consistent.
```suggestion
    title: Armorblox API Key
    description: Armorblox API Key
    type: credential_secret_key
```

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]

**Review by ablakley-r7 — CHANGES_REQUESTED** [review]
The plugin looks great. Outside of Igors and Connors suggestions I only have one.

File: plugins/armorblox/unit_test/test_get_remediation_action.py
Comment (SamhithaTatipalli) [reply plugins/armorblox/unit_test/test_get_remediation_action.py:1]: @cmcnally-r7 For some other plugins also we are not using parameterized, Is it a must to use parameterized?

File: plugins/armorblox/help.md (hunk 1 above)
Comment (SamhithaTatipalli) [reply plugins/armorblox/help.md:103]: @igorski-r7 Could you please suggest the format?

File: plugins/armorblox/unit_test/test_get_remediation_action.py
Comment (SamhithaTatipalli) [reply plugins/armorblox/unit_test/test_get_remediation_action.py:1]: @cmcnally-r7 , I have used parameterized in the test cases, some checks have not yet completed. Can you please verify it?

**Comment by SamhithaTatipalli** [comment]
@igorski-r7 , @cmcnally-r7 , @ablakley-r7 - code/synk check is pending... Could you please verify at your end and let me know?

**Review by cmcnally-r7 — COMMENTED** [review]
@SamhithaTatipalli From our end, the `synk` checks have run successfully. This is a bit of a bug that happens with the synk checks on github from time to time, and they don't show up as completed.

**Comment by SamhithaTatipalli** [comment]
@cmcnally-r7 , Okay...can I know the status of this PR?

**Review by cmcnally-r7 — COMMENTED** [review]
@SamhithaTatipalli We will need to re-review with your new changes now - we will make it a priority to get this looked at

File: plugins/armorblox/help.md [hunk 20]
```diff
@@ -0,0 +1,165 @@
...
+|Name|Type|Required|Description|
+|----|----|--------|-----------|
+|incidents|[]incident|True|A list of incidents identified by Armorblox|
+
+Example output:
+
+```
+{
+    "incidents": "some incidents"
```
Comment (igorski-r7) [inline plugins/armorblox/help.md:102]: Please pretty print it with 2 space indentation
```suggestion
  "incidents": "some incidents"
```

File: plugins/armorblox/icon_armorblox/util/constants.py [hunk 21]
```diff
@@ -0,0 +1,3 @@
+ARMORBLOX_INCIDENT_API_TIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+ARMORBLOX_INCIDENT_API_TIME_DELTA_IN_DAYS = 1
+DEFAULT_INTERVAL_VALUE = 1
```
Comment (igorski-r7) [inline plugins/armorblox/icon_armorblox/util/constants.py:3]: Please always set that value the same as `default` in plugin.spec.yaml file 😄 
```suggestion
DEFAULT_INTERVAL_VALUE = 600
```

File: plugins/armorblox/unit_test/test_get_remediation_action.py [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+
+
+@patch("requests.get", side_effect=Util.mocked_requests)
+class TestGetIndicatorDetails(TestCase):
+    @parameterized.expand([("10597"),("11081"),("11063")])
+    def test_get_remediation_action(self, mock_post, incident_id):
+        action = Util.default_connector(GetRemediationAction())
+        actual = action.run({Input.INCIDENT_ID: incident_id})
+        expected = {'remediation_details': 'WILL_AUTO_REMEDIATE'}
+        self.assertEqual(actual, expected)
```
Comment (igorski-r7) [inline plugins/armorblox/unit_test/test_get_remediation_action.py:23]: I think you could move that `default_connector` to the `def setUp(self)` 😄 
```suggestion
class TestGetIndicatorDetails(TestCase):
    def setUp(self) -> None:
        self.action = Util.default_connector(GetRemediationAction())
    @parameterized.expand([("10597"),("11081"),("11063")])
    def test_get_remediation_action(self, mock_post: Mock, incident_id: str) -> None:
        actual = self.action.run({Input.INCIDENT_ID: incident_id})
        expected = {'remediation_details': 'WILL_AUTO_REMEDIATE'}
        self.assertEqual(actual, expected)
```

**Review by cmcnally-r7 — APPROVED** [review]

**Comment by cmcnally-r7** [comment]
@SamhithaTatipalli Sorry for the delay on the `synk` checks. One last thing, if more review comments are addressed and you need us to re-review, please click the `Re-request review` button beside our names so we get email notified 😃

**Review by igorski-r7 — APPROVED** [review]

**Review by ablakley-r7 — APPROVED** [review]

**Comment by dlaverty-r7** [comment]
@SamhithaTatipalli , are you happy for us to go ahead and merge this PR to develop or have you any further updates you'd like to make beforehand?

File: plugins/armorblox/plugin.spec.yaml [hunk 23]
```diff
@@ -0,0 +1,206 @@
+plugin_spec_version: v2
+extension: plugin
+products: [insightconnect]
+name: armorblox
+title: Armorblox
+description: Armorblox is an API-based platform that stops targeted email attacks, protects sensitive data, and automates incident response
+version: 1.0.0
+vendor: armorblox
+author_email: app-publishers@armorblox.com
+support: rapid7
```
Comment (dlaverty-r7) [inline plugins/armorblox/plugin.spec.yaml:10]: Can you update the support field to:
support: community

**Review by dlaverty-r7 — APPROVED** [review]

**Comment by dlaverty-r7** [comment]
@SamhithaTatipalli , thanks for updating the support field. Are you happy for us to go ahead and merge this PR to develop now or have you any further updates you'd like to make beforehand?

**Pull request merged**